andela/neon-ah-backend
# #162414285 Bookmark articles

#### What does this PR do?
Create, Get and Remove Bookmarks on an Article
#### Description of Task to be completed?
 • Create Bookmark controller and routes
 • Create controller methods (`createOrRemove`, `read`) for performing actions on bookmark resource
 • Modify related model/migration files and associations
 • Write tests
 • Create a helper function to handle check for resource existence
 • Refactor bookmark controller using helper function

#### How should this be manually tested?
 • Clone this branch
 • CD into the repo
 • Run `npm install`
 • Run `npm run migrate && npm run seed`
 • Run `npm run dev`
 • Use postman to check the following endpoints:
```
 • http://localhost/3000/articles/:slug/bookmark (POST OR DELETE);
`Initial request creates bookmark, the next request removes the bookmark..and so on`
 • http://localhost/3000/articles/:slug/bookmarks (GET article's bookmarks);
 • http://localhost/3000/users/:userName/bookmarks (GET user's bookmarks); 
```

#### Background contexts?
Note that you must be an authenticated user to perform endpoint operations.
 • Login with the following credentials to access endpoints
 ``` 
 email: steve@now.com
 password: Blahblah
``` 
 • Use generated token as `bearer token` auth type to access endpoints.

#### What are the relevant pivotal tracker stories?
[Create, Get and Remove Article Bookmarks](https://www.pivotaltracker.com/story/show/162414285)
